utilities/dbviz/src/postgresql.rs

```diff
@@ -0,0 +1,312 @@
+                                match self.opts.column_description_wrap {
+                                    Some(wrap) => Some(textwrap::fill(&desc, wrap)),
+                                    None => Some(desc),
+                                }
+                            },
+                            None => None,
+                        };
+                        field.description = desc;
+
+                        field
+                    })
+                    .collect();
+
+                let desc = match &fields[0].table_description {
+                    Some(desc) => {
+                        match self.opts.table_description_wrap {
+                            Some(wrap) => Some(textwrap::fill(desc, wrap)),
+                            None => Some(desc).cloned(),
+                        }
+                    },
+                    None => None,
+                };
+
+                Table {
+                    name,
+                    description: desc,
+                    fields,
+                }
+            })
+            .collect();
+
+        let relations: Vec<_> = relations_rows
+            .into_iter()
+            .map(|row| {
+                let relation: Relation = row.try_into().unwrap();
+                relation
+            })
+            .filter(|relation| {
+                if self.include_table(&relation.on_table)
+                    && !self.exclude_table(&relation.on_table)
+                    && !self.exclude_table(&relation.to_table)
+                {
+                    if !self.include_table(&relation.to_table) {
+                        match partial_tables.get_mut(&relation.to_table) {
+                            Some(value) => {
+                                if !value.contains(&relation.to_field) {
+                                    value.push(relation.to_field.clone());
+                                }
+                            },
+                            None => {
+                                partial_tables.insert(relation.to_table.clone(), vec![relation
+                                    .to_field
+                                    .clone()]);
+                            },
+                        }
+                    }
+                    true
+                } else {
+                    false
+                }
+            })
+            .collect();
+
+        Ok(Schema {
+            tables,
+            relations,
+            partial_tables,
+        })
+    }
+}
+
+impl TryFrom<Row> for Index {
+    type Error = String;
+
+    fn try_from(row: Row) -> std::result::Result<Self, String> {
+        let all_fields: String = row.get(4);
+        let braces: &[_] = &['{', '}'];
+
+        let fields: Vec<_> = all_fields
+            .trim_matches(braces)
+            .split(',')
+            .map(std::string::ToString::to_string)
+            .collect();
+
+        Ok(Self {
+            table: row.get(0),
+            // name: row.get(1),
```

Comment:
Do we need this?

Reply:
No, but similar to above I am inclined to leave it in in-case Postgres changes how we can interogate the schema from it.
Its not very well documented internally.